threeML/plugins/Fermi_GBM/download_GBM_data.py

```diff
@@ -273,3 +344,241 @@ def cleanup_downloaded_GBM_data(detector_information_dict):
+    remote_files_info = DictWithPrettyPrint([(det, {}) for det in detectors])
+
+    # Classify the files detector by detector
+
+    tte_info = {}
+
+    for this_file in remote_file_list:
+
+        # this_file is something like glg_tte_n9_bn100101988_v00.fit
+        tokens = this_file.split("_")
+
+        if len(tokens) != 5:
+
+            # Not a data file
+
+            continue
+
+        else:
+
+            # The "map" is necessary to transform the tokens to normal string (instead of unicode),
+            # because u"b0" != "b0" as a key for a dictionary
+
+            _, file_type, detname, time_tag, version_ext = map(str, tokens)
+
+
+        try:
+            version, ext = version_ext.split(".")
+
+        except(ValueError):
+
+            # sometimes GBM randomly GZips the data... so we have to deal that
+
+            version,ext,fmt = version_ext.split('.')
+
+            compress_tte = False
+
+
+        # We do not care here about the other files (posthist, spechist and so on),
+        # nor about files which pertain to other detectors
+
+        if file_type not in ['cspec', 'tte'] or ext not in ['pha','fit'] or detname not in detectors:
+
+            continue
+
+        else:
+
+            if file_type == 'tte':
+
+                # the TTE files are broken into slices
+                # so we must account for that
+
+                if remote_files_info[detname][file_type] is None:
+
+                    remote_files_info[detname][file_type] = {}
+
+                remote_files_info[detname][file_type][time_tag] = this_file
+
+
+
+
+            else:
+
+                remote_files_info[detname][file_type] = this_file
+
+    # Now download the files
+
+    download_info = DictWithPrettyPrint([(det, DictWithPrettyPrint()) for det in detectors])
+
+    _warned_tte = False
+
+
+
+    for detector in remote_files_info.keys():
+
+
+
+        remote_detector_info = remote_files_info[detector]
+        local_detector_info = download_info[detector]
+
+        # Get CSPEC file
+
+        if (data_type is None) or (data_type == 'cspec'):
+
+            local_detector_info['cspec'] = downloader.download(remote_detector_info['cspec'], destination_directory,
+                                                               progress=True)
+
+        # Get TTE file (compressing it if requested)
+        if (data_type is None) or (data_type=='tte') and ('tte' in local_detector_info):
+
+            for time_tag in remote_detector_info['tte'].keys():
+
+                local_detector_info['tte'] = downloader.download(time_tag, destination_directory,
+                                                                 progress=True, compress=compress_tte)
+
+        else:
+            if not _warned_tte or data_type=='tte':
+
+                print('Continuous TTE data does not exist for this day')
+
+                _warned_tte = True
+
+    return download_info
+
+
+def _get_latest_version(filenames):
```

Comment:
This looks copy/pasted from download_GBM_data.py, if I am not mistaken. If possible please import it from there (or move it from there to a separate module as a function, and use it in both modules).

We need to avoid copying/pasting identical code as much as possible.